### PR TITLE
[#49987] Allow filtering only by Roles that grant `work_package_assigned`

### PR DESCRIPTION
**Pull request by aaron-contreras**
## Description
With the introduction of Work Package roles, these were included in the bundle of values for the "Assignee's role" filter. However, a deeper requirement was uncovered. We should only be displaying roles that allow becoming an assignee in this filter's values (those that include in its set of permissions `work_package_assigned`).

This means that the "Work package viewer" role should not be present in the list of available roles to filter by, but the "Work package commenter" and "Work package editor" should as they include in their permission set the `work_package_assigned`.

## Notes
See: https://community.openproject.org/work_packages/49987

### CODE REVIEW

File: app/models/queries/roles/filters/allows_becoming_assignee_filter.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  def allowed_values
+    [[I18n.t(:general_text_yes), OpenProject::Database::DB_VALUE_TRUE],
+     [I18n.t(:general_text_no), OpenProject::Database::DB_VALUE_FALSE]]
+  end
+
+  private
+
+  def assignable_permissions
+    RolePermission.where(permission: 'work_package_assigned')
+                  .select('id')
```
Comment (m-graf) [inline app/models/queries/roles/filters/allows_becoming_assignee_filter.rb:64]: Could `pluck` be used here? I think it's slightly more performant 😃

**Review by m-graf — APPROVED** [review]
COOL

File: app/models/queries/roles/filters/allows_becoming_assignee_filter.rb (hunk 1 above)
Comment (aaron-contreras) [reply app/models/queries/roles/filters/allows_becoming_assignee_filter.rb:64]: Using `pluck` in this context (subquery) results in an extra database call to resolve the IDs
```ruby
[18] pry(main)> Role.joins(:role_permissions).where("role_permissions.id IN (?)", RolePermission.where(permission: 'work_package_assigned').pluck('id'))

  RolePermission Pluck (1.0ms)
  
  SELECT "role_permissions"."id" 
  FROM "role_permissions" 
  WHERE "role_permissions"."permission" = $1  [["permission", "work_package_assigned"]]
  
  SQL (9.1ms) 
  
  SELECT "roles"."id" AS t0_r0, "roles"."name" AS t0_r1, "roles"."position" AS t0_r2, "roles"."builtin" AS t0_r3, "roles"."type" AS t0_r4, "roles"."created_at" AS t0_r5, "roles"."updated_at" AS t0_r6, "role_permissions"."id" AS t1_r0, "role_permissions"."permission" AS t1_r1, "role_permissions"."role_id" AS t1_r2, "role_permissions"."created_at" AS t1_r3, "role_permissions"."updated_at" AS t1_r4
  FROM "roles" 
  INNER JOIN "role_permissions" ON "role_permissions"."role_id" = "roles"."id"
  WHERE (role_permissions.id IN (9218,9344,9436,9448))
  ```

By using `select` the IDs don't need to be resolved in a separate query and results in a single SQL statement

```ruby
[19] pry(main)> Role.joins(:role_permissions).where("role_permissions.id IN (?)", RolePermission.where(permission: 'work_package_assigned').select('id'))

  SQL (4.7ms) 
  
  SELECT "roles"."id" AS t0_r0, "roles"."name" AS t0_r1, "roles"."position" AS t0_r2, "roles"."builtin" AS t0_r3, "roles"."type" AS t0_r4, "roles"."created_at" AS t0_r5, "roles"."updated_at" AS t0_r6, "role_permissions"."id" AS t1_r0, "role_permissions"."permission" AS t1_r1, "role_permissions"."role_id" AS t1_r2, "role_permissions"."created_at" AS t1_r3, "role_permissions"."updated_at" AS t1_r4
  FROM "roles" 
  INNER JOIN "role_permissions" ON "role_permissions"."role_id" = "roles"."id"
  WHERE (role_permissions.id IN (SELECT "role_permissions"."id"
                                                         FROM "role_permissions" 
                                                         WHERE "role_permissions"."permission" = 'work_package_assigned'))
  ```

Comment (m-graf) [reply app/models/queries/roles/filters/allows_becoming_assignee_filter.rb:64]: Fantastic.

```
                       .,,uod8B8bou,,.
              ..,uod8BBBBBBBBBBBBBBBBRPFT?l!i:.
         ,=m8BBBBBBBBBBBBBBBRPFT?!||||||||||||||
         !...:!TVBBBRPFT||||||||||!!^^""'   ||||
         !.......:!?|||||!!^^""'            ||||
         !.........||||                     ||||
         !.........||||  ##                 ||||
         !.........||||                     ||||
         !.........||||                     ||||
         !.........||||                     ||||
         !.........||||                     ||||
         `.........||||                    ,||||
          .;.......||||               _.-!!|||||
   .,uodWBBBBb.....||||       _.-!!|||||||||!:'
!YBBBBBBBBBBBBBBb..!|||:..-!!|||||||!iof68BBBBBb....
!..YBBBBBBBBBBBBBBb!!||||||||!iof68BBBBBBRPFT?!::   `.
!....YBBBBBBBBBBBBBBbaaitf68BBBBBBRPFT?!:::::::::     `.
!......YBBBBBBBBBBBBBBBBBBBRPFT?!::::::;:!^"`;:::       `.
!........YBBBBBBBBBBRPFT?!::::::::::^''...::::::;         iBBbo.
`..........YBRPFT?!::::::::::::::::::::::::;iof68bo.      WBBBBbo.
  `..........:::::::::::::::::::::::;iof688888888888b.     `YBBBP^'
    `........::::::::::::::::;iof688888888888888888888b.     `
      `......:::::::::;iof688888888888888888888888888888b.
        `....:::;iof688888888888888888888888888888888899fT!
          `..::!8888888888888888888888888888888899fT|!^"'
            `' !!988888888888888888888888899fT|!^"'
                `!!8888888888888888899fT|!^"'
                  `!988888888899fT|!^"'
                    `!9899fT|!^"'
                      `!^"'
```

**Review by klaustopher — APPROVED** [review]

**Pull request merged**